frontend/src/modules/Tables/components/TableFiltersDataGrid.js

```diff
@@ -0,0 +1,157 @@
+import { Card, CardContent, CircularProgress, Typography } from '@mui/material';
+import { DataGrid, GridActionsCellItem } from '@mui/x-data-grid';
+import { Warning } from '@mui/icons-material';
+import PropTypes from 'prop-types';
+import React, { useState } from 'react';
+
+import { DeleteObjectWithFrictionModal } from 'design';
+
+import DeleteIcon from '@mui/icons-material/DeleteOutlined';
+import { useTheme } from '@mui/styles';
+
+export const TableFiltersDataGrid = ({
+  items,
+  filter,
+  loading,
+  handlePageSizeChange,
+  handlePageChange,
+  deleteFunction,
+  assignFunction,
+  ...other
+}) => {
+  const theme = useTheme();
+  const [isDeleteFilterModalOpenId, setIsDeleteFilterModalOpen] = useState(0);
+
+  const handleDeleteFilterModalOpen = (id) => {
+    setIsDeleteFilterModalOpen(id);
+  };
+  const handleDeleteFilterModalClosed = () => {
+    setIsDeleteFilterModalOpen(0);
+  };
+
+  if (loading) {
+    return <CircularProgress />;
+  }
+
+  return (
+    <DataGrid
+      autoHeight
+      sx={{
+        wordWrap: 'break-word',
+        boxShadow: 2,
+        borderRadius: 1,
```

Comment:
I ran into the same issue when implementing datagrids for Redshift. For our dark theme the borders look bad. I am guessing that's why you added the borders here, but i think there is something off with the grey shadows and the header vs normal rows.
![image](https://github.com/user-attachments/assets/8aacf961-62dd-4b3b-9e8f-ffd9a8066c66)

I think that we should include the data grid style in the top-theme but for the time being here are the params that I used.

![image](https://github.com/user-attachments/assets/3724f4f2-c5ba-4bf2-9d53-7fd5562c4d56)




```
              sx={{
                wordWrap: 'break-word', //TODO: create a generic styled datagrid to be used across features
                '& .MuiDataGrid-row': {
                  borderBottom: '1px solid rgba(145, 158, 171, 0.24)'
                },
                '& .MuiDataGrid-columnHeaders': {
                  borderBottom: 0.5
                },
                '&.MuiDataGrid-root--densityStandard .MuiDataGrid-cell': {
                  py: '15px'
                }
              }}
```

Reply:
update DataGrid style to the above ++ hover piece because I think is a nice touch